### ISSUE TITLE
Hyphenation in the man page

### ISSUE THREAD

**Issue by d-e-s-o** [d-e-s-o/nitrocli#131]
Here is what a certain page of the manual currently looks like:
![pdf](https://user-images.githubusercontent.com/6610056/95018182-e2d94580-0612-11eb-915d-ff666e944390.png)

To me, it's confusing and actively wrong to hyphenate "code snippets" (whatever you may call them) the way it is happening. I even saw `NITROCLI_` being hyphenated.

I found [this](https://www.gnu.org/software/groff/manual/html_node/Manipulating-Hyphenation.html) manual page on how to control hyphenation.

It seems we could actually define hyphenation for words *including* their formatting:
```groff
.hw \fBnitrocli\fR \fB\-\-login\fR
```
And it appears as if that may be working. But of course...oh this pain. Maintaining a list of all bold/italic words... And then there is the issue that `\fB\-\-no\-capslock\fR` actually has a more or less natural hyphenation point (at the third hyphen), but that won't be displayed correctly:
![hyphen](https://user-images.githubusercontent.com/6610056/95018618-c4c11480-0615-11eb-882a-4882a9a064e1.png)

Any other ideas? I think we could hack together infrastructure that generates a list of all bold/italic words as part of `make doc` and includes them from the "main" man page file.

**Comment by d-e-s-o** [comment]
Oh, well, [it seems](https://man7.org/linux/man-pages/man7/groff_man_style.7.html) that we can use `\%` for that purpose.
```groff
\%\fB\-\-login\fR
```

That may be good enough then. I wonder when we reach a point where the actual text can no longer be deciphered from all the cryptic !#&*(!@# surrounding it.

**Comment by robinkrahl** [comment]
Hmm, good point.  I don’t know a solution for this problem offhand.

By the way, I already planned to open a pull request with some changes
to the man page after the current requests have landed:
- Even worse than this wrapping problem are the code examples.  My
  suggestion would be to hard-wrap them at 80 characters because this is
  a typical width for man viewers and it should also work for the PS/PDF
  output.
- groff recommends that sentences are separated by multiple spaces or by
  a newline (see groff(7)).  This enables cool features like better
  spacing when filling lines.

**Comment by d-e-s-o** [comment]
> - Even worse than this wrapping problem are the code examples. My suggestion would be to hard-wrap them at 80 characters because this is a typical width for man viewers and it should also work for the PS/PDF output.

Yeah. I was just experimenting with `.EX` & `.EE` but they just blow through any margins:
![screen](https://user-images.githubusercontent.com/6610056/95019201-dc4dcc80-0618-11eb-9c0a-f18831223276.png)
(first two are wrapped in these macros, the third isn't)

> - groff recommends that sentences are separated by multiple spaces or by a newline (see groff(7)). This enables cool features like better spacing when filling lines.

Interesting. Which section is that in? Can't find it off hand.

Perhaps we should get back to the initial discussion of whether we should really be writing in groff directly. If there was a neat higher level language that would just spit out that stuff and take care of the fine details itself, that would be my preferred choice.

**Comment by robinkrahl** [comment]
> > - groff recommends that sentences are separated by multiple spaces
> > or by a newline (see groff(7)). This enables cool features like
> > better spacing when filling lines.
> 
> Interesting. Which section is that in? Can't find it off hand.

Control characters, the items for ., space and newline.

> Perhaps we should get back to the initial discussion of whether we
> should really be writing in groff directly. If there was a neat higher
> level language that would just spit out that stuff and take care of
> the fine details itself, that would be my preferred choice.

There are two viable options that I’m aware of: pod2man is written in
and shipped with perl, scdoc is written in C and packaged for most
distributions.

Here are the respective man pages in their own format:
	https://perldoc.perl.org/pod2man.txt (at the end of the file,
	search for =head1)
	https://git.sr.ht/~sircmpwn/scdoc/blob/master/scdoc.5.scd

Both would make the file easier to read, but the question is whether
they would help us with the issues that actually hurt us, like wrapping
code snippets.

**Comment by robinkrahl** [comment]
I had another look at the two options.  As far as I see, `scdoc` does not support manipulating the hyphenation settings.  `pod2man` just disables hyphenation for the entire manpage.  The result does not look as bad as I thought, but I’d still prefer proper hyphenation if possible.

By the way, would you consider automatically generating the PDF file (and possibly an HTML version?) and pushing it to some webspace, e. g. GitHub Pages?  Shouldn’t be too difficult with GitHub Actions.

**Comment by d-e-s-o** [comment]
Yeah, no hyphenation may be better than what we currently have.

> By the way, would you consider automatically generating the PDF file (and possibly an HTML version?) and pushing it to some webspace, e. g. GitHub Pages? Shouldn’t be too difficult with GitHub Actions.

Yeah, we could do that. Ideally it would at least use the same commands / Makefile magic as we would use locally.
